int32 frequency
---
float64 gx
float64 gy
